napari/utils/theme.py

```diff
@@ -13,64 +16,117 @@
     use_gradients = False
 
 from ..utils.translations import trans
+from .events import EventedModel
+from .events.containers._evented_dict import EventedDict
 
 
-def __getattr__(attr):
-    if attr == "palettes":
-        warnings.warn(
-            trans._(
-                "palette is deprecated and will be removed after version 0.4.6. Please use get_theme and register_theme instead",
-                deferred=True,
-            ),
-            category=FutureWarning,
-            stacklevel=2,
+class Theme(EventedModel):
+    """Theme model.
+
+    Attributes
+    ----------
+    folder : str
+        Name of the virtual folder where icons will be saved to.
+    syntax_style : str
+        Name of the console style.
+        See for more details: https://pygments.org/docs/styles/
+    canvas : Color
+        Background color of the canvas.
+    background : Color
+        Color of the application background.
+    foreground : Color
+        Color to contrast with the background.
+    primary : Color
+        Color used to make part of a widget more visible.
+    secondary : Color
+        Alternative color used to make part of a widget more visible.
+    highlight : Color
+        Color used to highlight visual element.
+    text : Color
+        Color used to display text.
+    warning : Color
+        Color used to indicate something is wrong.
+    current : Color
+        Color used to highlight Qt widget.
+    """
+
+    folder: str
+    syntax_style: str
+    canvas: Color
+    console: Color
+    background: Color
+    foreground: Color
+    primary: Color
+    secondary: Color
+    highlight: Color
+    text: Color
+    icon: Color
+    warning: Color
+    current: Color
+
+    @validator("syntax_style", pre=True)
+    def _ensure_syntax_style(value: str) -> str:
+        from pygments.styles import STYLE_MAP
+
+        assert value in STYLE_MAP, (
+            "Incorrect `syntax_style` value provided. Please use one of the following:"
+            f" {', '.join(STYLE_MAP)}"
         )
-        return _themes
-    raise AttributeError
-
-
-_themes = {
-    'dark': {
-        'folder': 'dark',
-        'background': 'rgb(38, 41, 48)',
-        'foreground': 'rgb(65, 72, 81)',
-        'primary': 'rgb(90, 98, 108)',
-        'secondary': 'rgb(134, 142, 147)',
-        'highlight': 'rgb(106, 115, 128)',
-        'text': 'rgb(240, 241, 242)',
-        'icon': 'rgb(209, 210, 212)',
-        'warning': 'rgb(153, 18, 31)',
-        'current': 'rgb(0, 122, 204)',
-        'syntax_style': 'native',
-        'console': 'rgb(0, 0, 0)',
-        'canvas': 'black',
+        return value
+
+
+_themes = EventedDict(
+    {
+        'dark': Theme(
+            **{
+                'folder': 'dark',
```

Comment:
See comment above - but if we rename `folder` to `name`, is it weird if we then have a them with a different name in the `EventedDict` and in the `Theme` itself? Might be nice to force these to be the same somehow. Thoughts @tlambert03?

Reply:
I actually torn on this because I can envisage a situation where a theme is just a variation of an existing theme e.g. `dark` so the icon color is the same and there should be no need to create a new version of the icons under a different  namespace.

Reply:
ok, that's fine